cfme/rest/gen_data.py

```diff
@@ -193,50 +152,24 @@ def _finished():
 
 
 def services(request, rest_api, a_provider, dialog, service_catalogs):
-    service_data(request, rest_api, a_provider, dialog, service_catalogs)
-
-    try:
-        services = [_ for _ in rest_api.collections.services]
-        services[0]
-    except IndexError:
-        raise Exception("No options are selected")
-
-    @request.addfinalizer
-    def _finished():
-        services = [_ for _ in rest_api.collections.services]
-        if len(services) != 0:
-            rest_api.collections.services.action.delete(*services)
+    new_service = service_data(request, rest_api, a_provider, dialog, service_catalogs)
+    rest_service = rest_api.collections.services.get(name=new_service['service_name'])
+    # tests expect iterable
+    return [rest_service]
```

Comment:
return and delete (in service_data) just the service that was created here, don't touch anything else